WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReaders/CardReaderSettingsViewModel.swift

```diff
@@ -44,63 +36,43 @@ final class CardReaderSettingsViewModel: ObservableObject {
         knownReaders = []
         foundReader = nil
         connectedReader = nil
-        timer = nil // TODO Remove
-
-        // TODO Remove - simulates a declining battery level with a timer
-        batteryTimer = Timer.scheduledTimer(timeInterval: 3, target: self, selector: #selector(dummyUpdateBattery), userInfo: nil, repeats: true)
-    }
-
-    deinit {
-        batteryTimer?.invalidate()
     }
 
     func startSearch() {
         activeAlert = .searching
 
         let siteID = ServiceLocator.stores.sessionManager.defaultStoreID ?? Int64.min
 
-        let action = CardPresentPaymentAction.startCardReaderDiscovery(siteID: siteID, onCompletion: { cardReaders in
-            // TODO. To be implemented
-            // Leaving these prints here to help test the PR
-            print("==== new card readers discovered begins ====")
-            print("new readers: ", cardReaders)
-            print("==== new card readers discovered ends   ====")
+        let action = CardPresentPaymentAction.startCardReaderDiscovery(siteID: siteID, onCompletion: { [weak self] cardReaders in
+            self?.discoveredReader(cardReaders: cardReaders)
         })
 
         ServiceLocator.stores.dispatch(action)
-
-        // TODO Remove - simulates searching with a timer
-        timer = Timer.scheduledTimer(timeInterval: 3, target: self, selector: #selector(dummyFoundReader), userInfo: nil, repeats: false)
     }
 
-    // TODO Remove
-    @objc private func dummyFoundReader() {
-        foundReader = CardReader(serialNumber: "CHB204909001234", batteryLevel: 0.885)
-        activeAlert = .foundReader
-    }
+    private func discoveredReader(cardReaders: [CardReader]) -> Void {
+        if activeAlert == .foundReader {
```

Comment:
The usual (swiftly) idiom for early returns is `guard`. To word this differently, I think most iOS engineers would expect this to be:
`guard activeAlert != .foundReader else { return }`

Personally, I am not sure that's actually more readable, but I guess it's kind of the standard?

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/3784/commits/b7384077caaf1454e3c5bd58550704a402817399 (among other things)